include/psen_scan/udp_client.h

```diff
@@ -172,19 +182,18 @@ inline void UdpClient::sendCompleteHandler(const boost::system::error_code& erro
   std::cout << "Data successfully send." << std::endl;
 }
 
-template <std::size_t NumberOfBytesToSend>
-inline void UdpClient::write(const RawDataContainer<NumberOfBytesToSend>& data)
+inline void UdpClientImpl::write(std::shared_ptr<char> data, const std::size_t& number_of_bytes)
 {
-  io_service_.post([this, data]() {
-    socket_.async_send(boost::asio::buffer(data, data.size()),
-                       boost::bind(&UdpClient::sendCompleteHandler,
+  io_service_.post([this, data, number_of_bytes]() {
+    socket_.async_send(boost::asio::buffer(data.get(), number_of_bytes),
```

Comment:
why has this changed? from `data.size()` to `number_of_bytes`

Reply:
`data` is not an `std::array` anymore but a `char*`